test/test_00_basic.py

```diff
@@ -433,3 +433,24 @@ def test_FromVoidP_AsVoidP(self):
             @INIT
         """)
         assert mod.f(42) == 42
+
+    def test_leave_python(self):
+        mod = self.make_module("""
+            HPyDef_METH(f, "f", f_impl, HPyFunc_O)
+            static HPy f_impl(HPyContext *ctx, HPy self, HPy arg)
+            {
+                HPy_ssize_t data_len;
+                const char* data = HPyUnicode_AsUTF8AndSize(ctx, arg, &data_len);
+                HPy_ssize_t acount = 0;
+                HPy_BEGIN_LEAVE_PYTHON(ctx);
+                for (HPy_ssize_t i = 0; i < data_len; ++i) {
+                    if (data[i] == 'a')
+                        acount++;
+                }
+                HPy_END_LEAVE_PYTHON(ctx);
+                return HPyLong_FromSize_t(ctx, acount);
+            }
+            @EXPORT(f)
+            @INIT
+        """)
+        assert mod.f("abraka") == 3
```

Comment:
This is an OK sanity test but we could do better.
We need to check that we're not crashing if another thread mutates the variable which stores the string "abraka" and then we should run the test multiple times to try to trigger a crash.
Should I open an issue about it and we'll get to it later?

Reply:
yes, good suggestion